services/user/service.go

```diff
@@ -0,0 +1,30 @@
+package userService
+
+import (
+	"Bete/services/database"
+	"go.uber.org/fx"
+	"gorm.io/gorm"
+)
+
+type userService struct {
+	db *gorm.DB
+}
+
+type params struct {
+	fx.In
+
+	DbService database.Service
+}
+
+func New(p params) Service {
+	return &userService{
+		db: p.DbService.GetDBInstance(),
+	}
+}
+
+func (s userService) CreateUser(p CreateUserParams) User {
```

Comment:
nit let us call it EnsureUser

Reply:
I was thinking of renaming it to `ValidateOrCreateUser`

Reply:
I think the ensure prefix is a standard and more accurate